codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerProtocolHandshakeHandler.java

```diff
@@ -115,7 +115,15 @@ public void operationComplete(ChannelFuture future) {
 
     private boolean isNotWebSocketPath(FullHttpRequest req) {
         String websocketPath = serverConfig.websocketPath();
-        return serverConfig.checkStartsWith() ? !req.uri().startsWith(websocketPath) : !req.uri().equals(websocketPath);
+        boolean checkNextUri = true;
+        if (req.uri().length() > websocketPath.length()) {
+            char nextUri = req.uri().charAt(websocketPath.length());
+            if (nextUri != '/' && nextUri != '?') {
+                checkNextUri = false;
+            }
+        }
```

Comment:
All this should be done only in case of `serverConfig.checkStartsWith()`. In all other cases, we don't need to perform those checks.

Reply:
So..what should I do? That logic could be moved to new method?

Reply:
Probably own method makes sense as it became big enough.

Also, I think it would make sense to rename **isNotWebSocketPath** to `isWebSocketPath`. It's a generic Java way. When **NO** is required you can use **!isWebSocketPath()**.

Reply:
Thank you for your kind explanation.

I'll change it based on your reference.